services/src/main/java/com/linkedin/kmf/services/ProduceService.java

```diff
@@ -330,4 +331,18 @@ public void run() {
     }
   }
 
+  private class ProduceServiceThreadFactory implements ThreadFactory {
+
+    private final AtomicInteger _threadId = new AtomicInteger();
+    public Thread newThread(Runnable r) {
+      return new Thread(r, _name + " produce-service, produce thread " + _threadId.getAndIncrement());
+    }
+  }
+
+  private class HandleNewPartitionsThreadFactory implements ThreadFactory {
+    public Thread newThread(Runnable r) {
+      return new Thread(r, _name + " produce-service, handle-new-partitions thread ");
```

Comment:
I've removed "thread" from the thread name.  The thread runs the thing that handles new partitions and there are other types of threads that run in the produce service.